### PR TITLE
Ensure gp_hashagg_default_nbatches is a power of two.

### PR DESCRIPTION
**Pull request by hardikar**
This GUC should be a power of 2 for the proper functioning of HashAgg. It is not discoverable, but can be changed by customers - so this check is important.

shardikar=# set gp_hashagg_default_nbatches = 32;
SET
shardikar=# set gp_hashagg_default_nbatches = 33;
ERROR:  invalid value for parameter "gp_hashagg_default_nbatches": 33

@foyzur @hsyuan @karthijrk Please take a look.

### CODE REVIEW

File: src/backend/utils/misc/guc_gp.c [hunk 1]
```diff
@@ -5838,6 +5841,26 @@ assign_dispatch_log_stats(bool newval, bool doit, GucSource source)
 	return true;
 }
 
+bool
+assign_gp_hashagg_default_nbatches(int newval, bool doit, GucSource source)
+{
+	/* Must be a power of two */
```
Comment (hsyuan) [inline src/backend/utils/misc/guc_gp.c]: Will minus value come here?

Comment (hardikar) [inline src/backend/utils/misc/guc_gp.c]: Nope, that's checked with the min/max values of the GUC entries.
```
shardikar=# set gp_hashagg_default_nbatches = -20;
ERROR:  -20 is outside the valid range for parameter "gp_hashagg_default_nbatches" (4 .. 1048576)
```

Comment (hsyuan) [inline src/backend/utils/misc/guc_gp.c]: If it is not power of 2, can we show error messages "Must be a power of two."?

Comment (hardikar) [inline src/backend/utils/misc/guc_gp.c]: I'm not sure if there is a mechanism to do this apart from elog(ERROR). Let me check with @foyzur .

**Comment by hardikar** [comment]
@hsyuan This is error message instead : 
```
shardikar=# set gp_hashagg_default_nbatches = 33;
ERROR:  gp_hashagg_default_nbatches must be a power of two: 33
```

**Pull request merged**